src/utils/PWA.tsx

```
@@ -0,0 +1,166 @@
+ * mobile devices that are not already in standalone mode. On iOS devices, it
+ * will display a message explaining how to install the app. Otherwise, it will
+ * display a button to trigger the browser's install prompt.
+ * @returns {JSX.Element} The rendered component.
+ */
+export function InstallPrompt() {
+  const [isIOS, setIsIOS] = useState(false);
+  const [isStandalone, setIsStandalone] = useState(false);
+
+  useEffect(() => {
+    setIsIOS(
+      /iPad|iPhone|iPod/.test(navigator.userAgent) && !(window as any).MSStream,
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)